### DISCUSSION TITLE
Multiple conditions within on_finish

### DISCUSSION THREAD

**Discussion by blueraindrops** [jspsych/jsPsych#1878] [Q&A]
Hi everyone, in my code, I have a feedback message after each trial where two choices (A and B) are displayed randomly left or right. You can press either the 'z' (left) or the 'm' (right) key to choose either option. In the feedback message, I want to display what was chosen by comparing the positions of the choices and the response key using `on_finish`. 

For example, if A was displayed left during the trial, and B was displayed right, this is what stimulus would say:

```javascript
var a_position = 'left';
var b_position = 'right';
var a_key = 'z';
```
And vice versa if A and B switched positions. And then using on_finish:

```javascript
on_finish: function(data) {
    if(jsPsych.pluginAPI.compareKeys(data.response, a_key)){
      data.chose_a = true;
    } else {
      data.chose_a = false; 
    }
}
```
I then use `jsPsych.data.get().last(1).values()[0].response` to display the feedback message. However, this is not working because I don't think `jsPsych.pluginAPI.compareKeys` allows comparisons to a variable, only a defined string, even if the variable contains a string. 

I've also tried using multiple conditions within on_finish (for example, `if(jsPsych.pluginAPI.compareKeys(data.response, 'z') && a_position == 'left'`) but this doesn't seem to work either. Is there any way I can use on_finish to satisfy multiple conditions when displaying a feedback message? Thank you in advance!

**Comment by cslongman** [comment]
Looks to me like you might need to pass the relevant variables for comparison into the timeline node too.

You do that using something like...

```data: {'a_position': a_position}```

Hope this helps,

Cai
